test/Microsoft_Tests/src/SQLServer_Spec.enso

```diff
@@ -0,0 +1,150 @@
+        row3 = ["def", 42, True, 1.4, 1.4, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing]
+        table = Table.from_rows ["strs", "ints", "bools", "reals", "floats", "tinyints", "smallints", "bigints", "times", "dates", "datetimes", "smalldatetimes", "datetime2s", "datetimeoffsets"] [row1, row2, row3]
+        table2 = table.cast ['tinyints'] (Value_Type.Integer Bits.Bits_16)
+        table3 = table2.cast ['smallints'] (Value_Type.Integer Bits.Bits_16)
+        table4 = table3.cast ['ints'] (Value_Type.Integer Bits.Bits_32)
+        Panic.rethrow <|
+            t.update_rows table4 update_action=Update_Action.Insert
+
+        [connection, tinfo, t]
+
+    teardown self =
+        self.connection.execute_update 'DROP TABLE "'+self.tinfo+'"'
+        self.connection.close
+
+get_configured_connection_details =
+    host = Environment.get "ENSO_SQLSERVER_HOST" if_missing="localhost"
+    port = Environment.get "ENSO_SQLSERVER_PORT" if_missing="1433"
+    user = Environment.get "ENSO_SQLSERVER_USER" if_missing="sa"
+    password = Environment.get "ENSO_SQLSERVER_PASSWORD" if_missing="<YourStrong@Passw0rd>"
+    database = Environment.get "ENSO_SQLSERVER_DATABASE"
+    resolved_password = if password.starts_with "enso://" then Enso_Secret.get password else password
+    credentials = Credentials.Username_And_Password user resolved_password
+    if database.is_nothing then Nothing else
+        SQLServer_Details.SQLServer host port credentials database
+
+## Returns a function that takes anything and returns a new connection.
+   The function creates a _new_ connection on each invocation
+   (this is needed for some tests that need multiple distinct connections).
+create_connection_builder =
+    connection_details = get_configured_connection_details
+    connection_details.if_not_nothing <|
+        _ -> Database.connect connection_details
+
+add_specs suite_builder =
+    case create_connection_builder of
+        Nothing ->
+            message = "SQLServer test database is not configured. See README.md for instructions."
+            suite_builder.group "[SQLServer] Database tests" pending=message (_-> Nothing)
+        connection_builder ->
+            suite_builder.group "[SQLServer] Small" group_builder->
+                group_builder.specify "debugtest" <|
+                    connection = Database.connect get_configured_connection_details
+                    tinfo = Name_Generator.random_name "Tinfo"
+                    connection.execute_update 'Create Table "'+tinfo+'" ("times" TIME)'
+                    t = connection.query (SQL_Query.Table_Name tinfo)
+                    row1 = [Time_Of_Day.new 12 12 12 1 1 1]
+                    row2 = [Time_Of_Day.new 7 12 12 1 1 1]
+                    row3 = [ Nothing]
+                    table = Table.from_rows ["times"] [row1, row2, row3]
+                    Panic.rethrow <|
+                        t.update_rows table update_action=Update_Action.Insert
```

Comment:
We should try to clean-up this table.

Our server runs in Docker image so it is temporary. But if someone runs it on some permanent local deployment, they will be getting more and more trash tables with each run.